### PR TITLE
Fixed the functionality of START_GAME Event

### PR DESCRIPTION
**Pull request by Abhishek-Punhani**
## Description
 Added handler for StartGame at frontend and made some minor corrections. Also Triggered Sync_State Event after Start_Game and draw_card event as these events make changes using backend functions.
These changes require to synced just after the event is triggered
## Checklist
- [ ] I have tested these changes locally.
- [ ] I have reviewed the code and ensured it follows the project's coding guidelines.
- [ ] I have updated the documentation, if necessary.
- [ ] I have assigned reviewers to this pull request.

## Screenshots (if applicable)
[If your changes include any visual updates, provide screenshots here.]

### CODE REVIEW

File: frontend/src/pages/Game.tsx [hunk 1]
```diff
@@ -78,10 +80,11 @@ function Game() {
         const shuffledCards = [...cards].sort(() => Math.random() - 0.5);
         return shuffledCards.slice(0, 7);
     }, []);
+    console.log(gameState);
     useEffect(() => {
         modal.show(<GamePropertiesModal />, 'large', [], false);
         // eslint-disable-next-line
-    }, []); // todo add the required dependencies
+    }, [gameState.players.length, gameState.id]); // todo add the required dependencies
```
Comment (kuv2707) [inline frontend/src/pages/Game.tsx:87]: Including id here is necessary as previously explained. As long as the Game component is mounted, the game's id wont change.

This change will cause a UX inconvenience as previously mentioned, but we can ignore that for now.

File: frontend/src/pages/Game.tsx [hunk 2]
```diff
@@ -78,10 +80,11 @@ function Game() {
         const shuffledCards = [...cards].sort(() => Math.random() - 0.5);
         return shuffledCards.slice(0, 7);
     }, []);
+    console.log(gameState);
```
Comment (kuv2707) [inline frontend/src/pages/Game.tsx:83]: Remove this

**Review by kuv2707 — COMMENTED** [review]
Left some suggestions. Looks good mostly, haven't tested menually

File: frontend/src/pages/Game.tsx (hunk 1 above)
Comment (Abhishek-Punhani) [reply frontend/src/pages/Game.tsx:86]: Without this , for joining players game id is not visible on the modal , so i included it , for now, i guess, we can keep it like this and fix this in upcoming prs.

**Review by kuv2707 — APPROVED** [review]
Seems good

**Pull request merged**